src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -221,6 +227,14 @@ public void unsubscribe() {
 
   @Override
   public ConsumerRecords<K, V> poll(long timeout) {
+    if (_lastProcessedResult != null && _lastProcessedResult.exception() != null) {
```

Comment:
since we throw the exception from processing batch N-1 on the Nth call to poll(), wouldnt an "early" call to commit + crash before Nth poll() call make us skip an entire batch ?